packages/core/src/Parser.ts

```diff
@@ -217,33 +221,34 @@ export class Parser {
             node.nextSibling.nodeName === 'BR' &&
             !node.nextSibling.nextSibling
         ) {
-            context.node = node.nextSibling;
-            VDocumentMap.set(parsedNode, context.node);
+            context.currentNode = node.nextSibling;
+            VDocumentMap.set(parsedNode, context.currentNode);
         }
         this.currentContext = context;
     }
     /**
      * Parse the given text node into VNode(s).
      */
     _parseTextNode(): void {
-        const node = this.currentContext.node;
+        const node = this.currentContext.currentNode;
         const parentVNode = this.currentContext.parentVNode;
         const format = this.currentContext.format[0];
         this.parseNode();
-        const parsedVNodes = this.currentContext.lastParsed as CharNode[];
-        parsedVNodes.forEach((parsedVNode: CharNode, index: number) => {
-            parsedVNode.format = { ...parsedVNode.format, ...format };
-            VDocumentMap.set(parsedVNode, node, index);
-            parentVNode.append(parsedVNode);
-        });
+        Array.from(this.currentContext.parsingMap.keys()).forEach(
```

Comment:
The context being somewhat hidden in `this` is particulary hurting here, because we have no way to tell that calling `this.parseNode()` had any impact on `this.currentContext.parsingMap` !